packages/devextreme/js/core/utils/date.js

```diff
@@ -689,6 +689,45 @@ const getMachineTimezoneName = () => {
         : null;
 };
 
+const getRangesByDates = (dates) => {
```

Comment:
We calculate the same value for area-label three times: once for the visible view and twice for the invisible ones. Is there any way to optimize this and calculate the value only once?

Reply:
I don't think it's important, anyway we have O(n) time complexity

Reply:
We can, but for that I would introduce a new ranges option at the calendar level.